x/mongo/driver/topology/server.go

```diff
@@ -781,6 +803,85 @@ func (ss *ServerSubscription) Unsubscribe() error {
 	return nil
 }
 
+// publishes a ServerDescriptionChangedEvent to indicate the server description has changed
+func (s *Server) publishServerDescriptionChangedEvent(prev *description.Server, current *description.Server) {
```

Comment:
Can all of the event-publishing helpers that require server descriptions take a value rather than a pointer? The code that does the actual monitoring check keeps the descriptions as pointers because it needs to tell if it's been set, but that's an implementation detail that I'd like to avoid leaking to other helpers. Once we get to these helpers, we know the description has been set and can use it without taking a pointer.